### PR TITLE
Chat2 improvements: remove noise, set sender timestamp

### PR DESCRIPTION
**Pull request by jm-clius**
Two improvements for `chat2`:
1. Do not print messages that are not on the `chat2` content topic (especially important once message rates for other applications increase).
2. Set the message `timestamp` for `store` consistency.

### CODE REVIEW

**Review by staheri14 — APPROVED** [review]

File: examples/v2/chat2.nim [hunk 1]
```diff
@@ -216,7 +216,7 @@ proc publish(c: Chat, line: string) =
       encodedPayload = payload.encode(version, c.node.rng[])
     if encodedPayload.isOk():
       var message = WakuMessage(payload: encodedPayload.get(),
-        contentTopic: c.contentTopic, version: version)
+        contentTopic: c.contentTopic, version: version, timestamp: Timestamp(time*1000*1000*1000))
```
Comment (kaiserd) [inline examples/v2/chat2.nim:219]:
```suggestion
        contentTopic: c.contentTopic, version: version, timestamp: Timestamp(time*1000_000_000))
```

File: examples/v2/chat2.nim [hunk 2]
```diff
@@ -238,7 +238,7 @@ proc publish(c: Chat, line: string) =
   else:
     # No payload encoding/encryption from Waku
     var message = WakuMessage(payload: chat2pb.buffer,
-      contentTopic: c.contentTopic, version: 0)
+      contentTopic: c.contentTopic, version: 0, timestamp: Timestamp(time*1000*1000*1000))
```
Comment (kaiserd) [inline examples/v2/chat2.nim:241]:
```suggestion
      contentTopic: c.contentTopic, version: 0, timestamp: Timestamp(time*1000_000_000))
```

**Review by kaiserd — APPROVED** [review]
LGTM!

**Review by oskarth — APPROVED** [review]

File: examples/v2/chat2.nim (hunk 1 above)
Comment (s1fr0) [reply examples/v2/chat2.nim:219]: You can use getNanosecondTime() for the conversion (and omit Timestamp type specification):

```suggestion
        contentTopic: c.contentTopic, version: version, timestamp: getNanosecondTime(time))
```

File: examples/v2/chat2.nim (hunk 2 above)
Comment (s1fr0) [inline examples/v2/chat2.nim:241]: You can use getNanosecondTime() for the conversion (and omit Timestamp type specification):

```suggestion
      contentTopic: c.contentTopic, version: 0, timestamp: getNanosecondTime(time))
```

**Review by s1fr0 — COMMENTED** [review]
Looks good to me! Left some comments.

**Pull request merged**